c10/core/ConstantSymNodeImpl.cpp

```diff
@@ -11,7 +11,7 @@ namespace c10 {
 #define DEFINE_BINARY_OP(OP, ROP)                                        \
   template <typename T>                                                  \
   c10::SymNode ConstantSymNodeImpl<T>::OP(const c10::SymNode& other) {   \
-    TORCH_INTERNAL_ASSERT(other->singleton_int().has_value());           \
+    TORCH_INTERNAL_ASSERT(other->is_singleton());                        \
```

Comment:
Since this PR is so long, I'm obligated to mention that stuff like this could have easily been split off into its own, easy to land PRs.